src/fstab-generator/fstab-generator.c

```diff
@@ -936,6 +936,10 @@ static int parse_fstab_one(
                         mount_is_network(fstype, options) ? SPECIAL_REMOTE_FS_TARGET :
                                                             SPECIAL_LOCAL_FS_TARGET;
 
+        /* Setting nofail for /sysroot/usr doesn't make sense so let's just silently ignore it. */
+        if (is_sysroot_usr && FLAGS_SET(flags, MOUNT_NOFAIL))
```

Comment:
the right hand side of the && is redundant, since the &= expression below is just a NOP if the flag is off

Reply:
Well if I want to do 3. then I need to test which flag is set and log about it only if it is.